biosimulations/apps/dispatch-api/src/app/app.controller.ts

```diff
@@ -138,102 +127,66 @@ export class AppController implements OnApplicationBootstrap {
-            jsonResults[directoryName] = {};
-          }
-
-          const jsonPath = filePath.split('.csv')[0] + '.json';
-
-          const jsonResult: object[] = [];
-          setTimeout(() => {
-            fs.createReadStream(filePath)
-              .pipe(
-                csv2Json.default({
-                  separator: ',',
-                })
-              )
-              .pipe(fs.createWriteStream(jsonPath));
-          }, 0);
-
-          const fileContent = fs.readFileSync(jsonPath);
-          jsonResults[directoryName][taskName] = JSON.parse(
-            fileContent.toString()
-          );
+    sedmls.forEach((sedml) => {
+      structure[sedml] = [];
+      const taskFiles = fs.readdirSync(path.join(resultPath, sedml));
+      taskFiles.forEach((taskFile) => {
+        if (taskFile.endsWith('.csv')) {
+          structure[sedml].push(taskFile.split('.csv')[0]);
         }
       });
     });
+
     return {
-      message: 'Data fetched successfully',
-      data: chart ? this.convertJsonDataToChartData(jsonResults) : jsonResults,
+      message: 'OK',
+      data: structure,
     };
   }
 
-  convertJsonDataToChartData(data: any) {
-    const finalRes = JSON.parse(JSON.stringify(data));
-    const sedmls = Object.keys(data);
-    let tasks: Array<string> = [];
-    // sedmls.forEach(sedml => {
-    //   finalRes.set(sedml, {});
-    // })
-
-    sedmls.forEach((sedml) => {
-      tasks = Object.keys(data[sedml]);
-      tasks.forEach((task) => {
-        finalRes[sedml][task] = {};
-      });
-    });
-
-    for (const sedml of sedmls) {
-      for (const task of tasks) {
-        const taskKeys = Object.keys(data[sedml][task][0]);
-        taskKeys.splice(taskKeys.indexOf('time'), 1);
-
-        for (const taskKey of taskKeys) {
-          finalRes[sedml][task][taskKey] = {};
-          finalRes[sedml][task][taskKey]['x'] = [];
-          finalRes[sedml][task][taskKey]['y'] = [];
-          finalRes[sedml][task][taskKey]['type'] = 'scatter';
-        }
+  @Get('result/:uuid')
+  @ApiResponse({
+    status: 200,
+    description: 'Get Simulation Results',
+    type: Object,
+  })
+  async getVisualizationData(
+    @Param('uuid') uId: string,
+    @Query('chart') chart: boolean,
+    @Query('sedml') sedml: string,
+    @Query('task') task: string
+  ) {
+    const fileStorage = process.env.FILE_STORAGE || '';
 
-        for (const dataObj of data[sedml][task]) {
-          for (const taskKey of taskKeys) {
-            finalRes[sedml][task][taskKey]['x'].push(dataObj['time']);
-            finalRes[sedml][task][taskKey]['y'].push(dataObj[taskKey]);
-          }
-        }
-      }
-    }
+    const jsonPath = path.join(
+      fileStorage,
+      'simulations',
+      uId,
+      'out',
+      sedml,
+      task
+    );
+    const filePath = chart ? `${jsonPath}_chart.json` : `${jsonPath}.json`;
+    const fileContent = JSON.parse(fs.readFileSync(filePath).toString());
```

Comment:
Same here, we need to switch all of these to async

Reply:
Done